diff --git a/Data/Mod.hs b/Data/Mod.hs
index 5ccbef7..71b4c0a 100644
--- a/Data/Mod.hs
+++ b/Data/Mod.hs
@@ -352,10 +352,8 @@ invertModInternal
   :: Natural -- Value
   -> Natural -- Modulo
   -> Maybe Natural
-invertModInternal 0 1 = Just 0
 invertModInternal x m = case integerRecipMod# (toInteger x) m of
-  -- See https://gitlab.haskell.org/ghc/ghc/-/issues/26017
-  (# | () #) -> if m == 1 then Just 0 else Nothing
+  (# | () #) -> Nothing
   (# y | #)  -> Just y
 {-# INLINABLE invertModInternal #-}
 
diff --git a/Data/Mod/Word.hs b/Data/Mod/Word.hs
index 1d0657e..45ad23c 100644
--- a/Data/Mod/Word.hs
+++ b/Data/Mod/Word.hs
@@ -335,8 +335,6 @@ invertMod :: KnownNat m => Mod m -> Maybe (Mod m)
 invertMod mx@(Mod !x) = case natVal mx of
   NatJ#{}   -> tooLargeModulus
   NatS# 0## -> Nothing
-  -- See https://gitlab.haskell.org/ghc/ghc/-/issues/26017
-  NatS# 1## -> Just 0
   NatS# m#  -> Mod <$> invertModWord x (W# m#)
 
 invertModWord :: Word -> Word -> Maybe Word
diff --git a/test/Test.hs b/test/Test.hs
index c996fd2..8825efa 100644
--- a/test/Test.hs
+++ b/test/Test.hs
@@ -58,8 +58,6 @@ main = defaultMain $ testGroup "All" $
   [ testGroup "Mod 1" $
     testProperty "fromInteger"
       (fromIntegerProp (Proxy :: Proxy 1)) :
-    testProperty "invertMod"
-      (invertMod (0 :: Mod 1) == Just 0) :
     map lawsToTest (laws1 (Proxy :: Proxy (Mod 1)))
   , testMod(2310)
   , testMod(2^16-1)
@@ -105,8 +103,6 @@ main = defaultMain $ testGroup "All" $
   , testGroup "Word.Mod 1" $
     testProperty "fromInteger"
       (fromIntegerWordProp (Proxy :: Proxy 1)) :
-    testProperty "invertMod"
-      (Word.invertMod (0 :: Word.Mod 1) == Just 0) :
     map lawsToTest (laws1 (Proxy :: Proxy (Word.Mod 1)))
   , testMod(2310)
   , testMod(2^16-1)
